proofs/erc20/tmpl.k

```diff
@@ -97,4 +97,12 @@ module {MODULE}-SPEC
      andBool 0 <=Int CALL_DEPTH andBool CALL_DEPTH <Int 1024
      {REQUIRES}
 
+  // **TODO**: These semmingly unnecessary claims are to be an workaround for the K prover issue
+  // that unresolved functions are not fully evaluated in the normal semantics rule application steps.
+  // These should be dropped once the issue is fixed.
```

Comment:
@daejunpark I was looking at the issue on Friday, and I don't really understand how putting these rules back in help. At least in the proof for viper's balanceOf case, the problem seemed to be that the `chop` rule from verification.md 
```
rule chop(I) => I requires 0 <=Int I andBool I <Int pow256
```
simplified expression in the term, but not in the lhs of the claim itself. Your K PR also seems to suggest a fix to that issue. So why is this required?

Reply:
Also, just to be sure - do you want the chop to simplify the expression as much as possible? Would adding a concrete attribute to the rule above not work? At least for some the erc 20 proofs, this seemed to work, but it broke the sum-to-n proof.

Reply:
As I said, the normal semantics rule application failed to evaluate the unresolved terms that can be evaluated by applying the non-semantics rules, the spec rules in this case.

`[concrete]` attribute for the rule you mentioned should have no effect since the semantic rules cover that. Do NOT add that.